### PR TITLE
Feature/mui5

### PR DESCRIPTION
**Pull request by villuv**
## Related Issue

#55 

## Description

Compatibility with MUI5 alphas

## Related PRs

Same PR in upstream: https://github.com/mbrn/material-table/pull/2542

## Additional Notes

Probably not a good idea to merge before MUI5 is at least in beta. Also not thorougly tested.

### CODE REVIEW

**Comment by matthewoestreich** [comment]
Thank you SO much!!!!!! <3

**Comment by Domino987** [comment]
This is actually really important and the beta will be released really soon. Since I see no way to support both version at the same time, maybe we should release a next branch and support these two until we retired v4 support. 
The problem is since v33 since fade was renamed to alpha the date picker breaks and there is no good date picker until the v5 lab. So I am all in favor of supporting two versions until we move all the way to v5.

File: src/components/m-table-edit-cell.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ import * as React from "react";
 import PropTypes from "prop-types";
 import TableCell from "@material-ui/core/TableCell";
 import CircularProgress from "@material-ui/core/CircularProgress";
-import { fade } from "@material-ui/core/styles/colorManipulator";
-import withTheme from "@material-ui/core/styles/withTheme";
+import { alpha } from "@material-ui/core/styles/colorManipulator";
```
Comment (Domino987) [inline src/components/m-table-edit-cell.js]: These deep access are actually [forbidden](https://next.material-ui.com/guides/minimizing-bundle-size/#option-1), Can we shorten them?

File: src/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React from "react";
 import { defaultProps } from "./default-props";
 import { propTypes } from "./prop-types";
 import MaterialTable from "./material-table";
-import withStyles from "@material-ui/core/styles/withStyles";
+import withStyles from "@material-ui/styles/withStyles";
```
Comment (Domino987) [inline src/index.js]: I think we should stay with the core import to access the theme from the surroundings? https://next.material-ui.com/styles/basics/#using-the-theme-context

Comment (villuv) [reply src/index.js]: Yes, I agree that the imports need to be refactored in general. For this PR I tried to keep the changeset minimal towards upstream so I basically just renamed things where needed.

File: src/components/m-table-edit-cell.js (hunk 1 above)
Comment (matthewoestreich) [reply src/components/m-table-edit-cell.js]: I added a new eslint rule that should catch this moving forward.

https://github.com/material-table-core/core/blob/efb68b7a17d14f3ed482de457c01c01f055be658/.eslintrc#L4-L9

File: src/index.js (hunk 2 above)
Comment (MrWittman612) [reply src/index.js]: Import { withstyles } from '@material-ui/core' works pre [MaterialUI docs](https://material-ui.com/guides/typescript/#usage-of-withstyles)

**Comment by Domino987** [comment]
Can we get the merge conflicts resolved so we can get a next branch running?

**Comment by Domino987** [comment]
@villuv ping ;)

**Comment by villuv** [comment]
GH doesn't show me any details on the conflicts other than file names.

**Comment by Domino987** [comment]
Interesting. I will take s look

**Comment by Domino987** [comment]
Close in favor of #216

**Pull request closed without merging**